auth/data.js

```diff
@@ -78,6 +79,70 @@ Client.createClientLoader = function() {
   };
 };
 
+Client.createClientLoader2 = function() {
+  var Client = this;
+  return async (clientId) => {
+    try {
+      var client = await Client.load(clientId);
+    } catch (err) {
+      debug('Failed to load client: ' + clientId);
+      throw new Error("Failed to load client");
+    }
+    if (client.expires.getTime() < new Date().getTime()) {
+      throw new Error("Credentials expired!");
+    }
+    return {
+      clientId:     client.clientId,
+      accessToken:  client.accessToken,
+      scopes:       client.scopes
+    };
+  };
+};
+
+
+/** Create caching client loader */
```

Comment:
This awesome, but I can't help but feel like a more descriptive name would be helpful to someone reading the code in the future. createCachedClientLoader ? #nit ;p